src/tls.c

```diff
@@ -786,6 +787,21 @@ static int connTLSWrite(connection *conn_, const void *data, size_t data_len) {
     return ret;
 }
 
+static int connTLSWritev(connection *conn_, const struct iovec *iov, int iovcnt) {
+    if (iovcnt == 1) return connTLSWrite(conn_, iov[0].iov_base, iov[0].iov_len);
+
+    size_t cum = 0;
+    for (int i = 0; i < iovcnt; i++) cum += iov[i].iov_len;
+
+    char buf[cum];
```

Comment:
There is a limitation in `_writeToClient`, it makes sure that the total amount of bytes to send won't be greater than `NET_MAX_WRITES_PER_EVENT`== 64KB.

Reply:
ok, but the limitation and the allocation are in two different functions.
or worse, this is a generic infrastructure, it can't assume anything about how it'll be used.
i.d like to implement the fallback mechanism i described, you can set the threshold to 64k so that it'll never actually run, but then it could mean we have a bunch of untested code.

p.s. looking at _writeToClient, it doesn't actually grantee anything about the size, since what it does is stop adding more nodes when you passed the 64kb limit, but it does that only after we passed it. so if we have two linked list nodes, one of 1kb, and the other of 20mb, it'l queue both.
or am i missing something?

Reply:
make sense, I will make a new approach based on your idea.